src/librustc/ty/error.rs

```diff
@@ -250,6 +250,21 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
                     db.note("no two closures, even if identical, have the same type");
                     db.help("consider boxing your closure and/or using it as a trait object");
                 }
+                match (&values.found.sty, &values.expected.sty) { // Issue #53280
```

Comment:
It's nice to provide a bit of context in the comments without the need to open a browser :)

As a rule, I add a one sentence long description of what the code is doing (or rather, why), with the issue number at the end for _extra_ context.

Reply:
I agree that issue numbers are merely an extra reference for the curious software-archeologist and not a substitute for explanatory comments where explanatory comments are merited, but I feel like "use a float literal" is pretty self-explanatory? I do write explanatory comments where I think it necessary (recent examples: [1](https://github.com/rust-lang/rust/blame/0aa8d0320266b5579428312095fe49af05ada972/src/librustc/hir/lowering.rs#L1772-L1777) [2](https://github.com/rust-lang/rust/blame/0aa8d0320266b5579428312095fe49af05ada972/src/libsyntax/parse/parser.rs#L6047-L6049)).